Add cancel button for when user entry is being edited

JeaNugroho opened the pull request:
* Resolves: #22608

## Summary
Add a cancel button between checkmark button and three-dot button for when user entry is being edited.
### Before
![Screen Shot 2023-06-22 at 8 02 07 PM](https://github.com/nextcloud/server/assets/35510082/fddd8aa8-0fdf-4d52-993f-58f1023b9948)

### After
![Screen Shot 2023-06-21 at 10 40 21 PM](https://github.com/nextcloud/server/assets/35510082/e56dc22e-7e4c-4b9b-8a56-d43ef91fcab4)


## Potential next steps
[This comment](https://github.com/nextcloud/server/issues/22608#issuecomment-751663701) may be the next fix to make user experience more intuitive and straightforward. I personally would think that the user editing process would be like this if I were a new user to the app. I can certainly implement a way to make this happen, either in the same PR or in the next PR after this one's completed.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included or are not applicable
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

JeaNugroho committed:
Add cancel button for when user entry is being edited
Rebase add cancel button
Signed-off-by: JeaNugroho <jeanugroho@yahoo.com>

Fenn-CS commented on apps/settings/src/components/Users/UserRow.vue:
```diff
@@ -239,11 +239,15 @@
 				class="toggleUserActions">
-				<NcActions>
+				<NcActions :inline="2">
 					<NcActionButton icon="icon-checkmark"
 						:title="t('settings', 'Done')"
 						:aria-label="t('settings', 'Done')"
 						@click="editing = false" />
```
```suggestion
```

Perhaps? I mean the check and the cancel appears to do the same thing, so it's even more confusing or misleading because it could imply that the `check` is to validate the action.

We should probably remove it or make the check validate all "line edits"

JeaNugroho replied:
@Fenn-CS I agree with you here that this is one of the fixes. 

However, I have a rough thought on how to implement the check button click as a validation to all line edits. This would mean submit any changes to the user fields by clicking on that check button and omit the "submit individual field changes by clicking on the arrow button in each field" functionality. I can add this as another commit to this PR. If it turns out that the fix is way harder than I thought, then I'll resort to deleting the check button.

Would you say the same?

JeaNugroho replied:
@Fenn-CS I might think that it's best to follow your suggestion. I just realized there could be more complicated things happening while editing the user entries. For e.g., an authorization popup can appear while editing one of the fields. This fix you suggested gives a minimum change that wouldn't bother other features that I have yet to know.

I've also added an additional commit to omit the `:inline="2"` bind since we deleted the checkmark button. Please let me know if I can change any other things, otherwise I'm happy if this PR can be completed and changes can be merged. This'd my first PR for Nextcloud and open source contribution =)

Fenn-CS replied:
Thanks for you contribution @JeaNugroho !

This now makes better sense to me.

nfebe reviewed (changes requested):
Thank you so far! Left an inline comment.

JeaNugroho committed:
Update apps/settings/src/components/Users/UserRow.vue

Omit checkmark button since it has the same functionality as the cancel button - Editing User Entry

Co-authored-by: F. E Noel Nfebe <fenn25.fn@gmail.com>
Signed-off-by: Jean Benedict Nugroho <35510082+JeaNugroho@users.noreply.github.com>

JeaNugroho committed:
Omit bind property for more than one inside NcActions

Signed-off-by: JeaNugroho <jeanugroho@yahoo.com>

artonge reviewed (commented):
Watch out as the component has been heavily modified recently, so your change might need to be adapted.

https://github.com/nextcloud/server/pull/38839

nimishavijay reviewed (changes requested):
Super nice! :) Only one tiny thing to nitpick: All the icons in a column should be center-aligned ideally. So the buttons should be moved slightly to the right, so that the close icon is nicely aligned with all the edit icons above it :)

nfebe reviewed (commented):
This now looks ok to me however,

- There's a review about alignment which you should address
- Update the screenshot in the description to reflect the new changes

And we would be good to go! Thank you for your contribution :)

susnux commented:
This needs to be rebased as the @artonge said it was heavily modified.
The settings are not saved when pressing the check button "done" but when you save a field. Similar to all other settings on Nextcloud which are saved on change rather than that there is a "save" button.

JeaNugroho commented:
Have just rebased! I'll try to fix the alignment and update the original commit with the new screenshot of the layout. And then when any of the required CI checks fail, I hope someone can provide guidance for that 😊

@Fenn-CS @susnux @artonge @nimishavijay

JeaNugroho commented:
@Fenn-CS @artonge @susnux 
We now clearly know that we need to fix eslint and Node. I recall from a different PR I'm doing, we need to run `npm run build` to fix Node and run `npm run lint:fix` to fix eslint. So I don't mess this PR like the other one, in what sequence do I have to run these? And are there any other commands or things should I do before, after, or in between these two commands?

Fenn-CS commented:
@JeaNugroho  you should lint and then build ideally! 

However there are some merge conflicts in your PR, check the files tab closely. I would leave some inline comments to help you see them.

Fenn-CS commented on apps/settings/src/components/Users/UserRow.vue:
```diff
@@ -221,10 +221,36 @@
+				<div class="icon-checkmark" />
+				{{ feedbackMessage }}
+			</div>
+>>>>>>> 7a2938b01f3 (Add cancel button for when user entry is being edited)
```
```suggestion
```

Merge conflict, and deleting this line is not the solution though you want to work with perhaps your code editor to help you resolve the conflicts.

Fenn-CS commented on apps/settings/src/components/Users/UserRow.vue:
```diff
@@ -221,10 +221,36 @@
 		</div>
 
 		<div class="userActions">
+<<<<<<< HEAD
```
```suggestion
```

Another conflict marker

Fenn-CS commented on apps/settings/src/components/Users/UserRow.vue:
```diff
@@ -221,10 +221,36 @@
 				:actions="userActions"
 				:edit="true"
 				@update:edit="toggleEdit" />
+=======
```
```suggestion
```

Conflict.

JeaNugroho committed:
Omit conflict markers

Signed-off-by: JeaNugroho <jeanugroho@yahoo.com>

Pytal commented:
Thanks @JeaNugroho! We have some more modifications incoming, would it be fine with you if we revisit this after merging https://github.com/nextcloud/server/pull/39050 to prevent conflicts?

joshtrichards commented:
#39050 has been merged, @JeaNugroho. So feel free to proceed. :)
